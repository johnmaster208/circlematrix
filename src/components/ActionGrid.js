import React from 'react';
import PropTypes from 'prop-types';
import ActionCircle from './ActionCircle'

const ActionGrid = (props) => {
  const { items, onClick } = props
  return (
    <div id="Action-Grid">
      {
        items.map((item, index) => {
          return (
            <ActionCircle
              key={index}
              item={item}
              onClick={onClick}
              isOccupied={item.wasSelectedByMe === true || item.wasSelectedByOther === true}
            />
          );
        })
      }
    </div>
  );
}

ActionGrid.propTypes = {
  items: PropTypes.array.isRequired,
  onClick: PropTypes.func.isRequired,
}


export default ActionGrid;
