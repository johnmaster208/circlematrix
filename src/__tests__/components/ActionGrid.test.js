import React from 'react';
import ActionGrid from '../../components/ActionGrid';
import { shallow } from 'enzyme';

describe('ActionGrid component', () => {

    const props = {
        items: [],
        onClick: jest.fn()
    };

    it('Renders without crashing...', () => {
        const wrapper = shallow(
            <ActionGrid
                {...props } />
        );
        expect(wrapper).toMatchSnapshot();
    });

});