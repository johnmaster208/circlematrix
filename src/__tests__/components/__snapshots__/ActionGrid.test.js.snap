// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActionGrid component Renders without crashing... 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ActionGrid
    items={Array []}
    onClick={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [],
      "id": "Action-Grid",
    },
    "ref": null,
    "rendered": Array [],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [],
        "id": "Action-Grid",
      },
      "ref": null,
      "rendered": Array [],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
